internal/state/state.go

```diff
@@ -10,43 +10,109 @@ import (
 	"sync"
 	"time"
 
+	prom "github.com/prometheus/client_golang/prometheus"
 	"go.uber.org/zap"
 )
 
 // State keeps the discovery service state.
-type State struct {
-	clusters sync.Map
```

Comment:
I don't see the reason for this change, and I don't think it's a good change unless there're reasons (?).

Cluster creation is infrequent, while cluster access if frequent. Lock contention is not something we're looking for here.

Reply:
I feel if we do a simple counter (as atomic variable if needed?) or directly via Prometheus gauge? we should be able to track precisely when the clusters are getting created/removed (and same with affiliates?).